src/include/sof/audio/kpb.h

```diff
@@ -107,6 +107,16 @@ enum kpb_id {
 	KPB_HP,
 };
 
+struct hb {
+	enum buffer_state state; /**< state of the buffer */
+	void *start_addr; /**< buffer start address */
+	void *end_addr; /**< buffer end address */
+	void *w_ptr; /**< buffer write pointer */
+	void *r_ptr; /**< buffer read pointer */
+	struct hb *next; /**< next history buffer */
```

Comment:
Can you use a list here (see list.h)

Reply:
@lgirdwood, well I wanted to use it from start however list.h turned out to be quite bulky for such circular buffer. It starts iterating from buffer->next, requires to pass list_item, `list_for_item` stop condition is just `item->next` which is not sufficient for our use because we want to terminate loop   sometimes even when there is `buff->next`. Please check `kpb_init_draining` and the procedure of finding read pointer.